library/core/src/num/nonzero.rs

```diff
@@ -1009,6 +1009,12 @@ macro_rules! nonzero_integer {
             #[must_use = "this returns the result of the operation, \
                           without modifying the original"]
             #[inline]
+            #[requires({
+                !self.get().checked_mul(other.get()).is_some()
```

Comment:
```suggestion
                self.get().checked_mul(other.get()).is_some()
```
(My bad, I made a typo last time)